crates/bevy_webxr/src/lib.rs

```diff
@@ -66,6 +66,65 @@ fn sync_head_tf(
     }
 }
 
+fn sync_frustum(
+    xr_ctx: NonSend<WebXrContext>,
+    frame: NonSend<web_sys::XrFrame>,
+    mut left_q: Query<&mut Frustum, (With<LeftEyeMarker>, Without<RightEyeMarker>)>,
+    mut right_q: Query<&mut Frustum, (With<RightEyeMarker>, Without<LeftEyeMarker>)>,
+) {
+    let reference_space = &xr_ctx.space_info.0;
+    let viewer_pose = frame.get_viewer_pose(&reference_space).unwrap();
+
+    let mut left_frustum = left_q.single_mut();
+    let mut right_frustum = right_q.single_mut();
+
+    let views: Vec<web_sys::XrView> = viewer_pose
+        .views()
+        .iter()
+        .map(|view| view.unchecked_into::<web_sys::XrView>())
+        .collect();
+
+    let left_eye: &web_sys::XrView = views
+        .iter()
+        .find(|view| view.eye() == web_sys::XrEye::Left)
+        .unwrap();
+
+    let left_proj: Mat4 = left_eye.projection_matrix().xr_into();
+
+    let fov = fov_from_mat4(left_proj);
+
+    let tf: Transform = left_eye.transform().xr_into();
+
+    // TODO: actual near/far plane conversion
+    let projection = WebXrPerspectiveProjection {
+        fov,
+        far: 1000.0,
+        ..default()
+    };
+    let view_proj = projection.get_projection_matrix() * tf.compute_matrix().inverse();
+    *left_frustum = Frustum::from_view_projection(&view_proj, &tf.translation, &tf.back(), projection.far);
+
+    let right_eye: &web_sys::XrView = views
+        .iter()
+        .find(|view| view.eye() == web_sys::XrEye::Right)
+        .unwrap();
+
+    let right_proj: Mat4 = right_eye.projection_matrix().xr_into();
+
+    let fov = fov_from_mat4(right_proj);
+
+    let tf: Transform = right_eye.transform().xr_into();
+
+    // TODO: actual near/far plane conversion
+    let projection = WebXrPerspectiveProjection {
```

Comment:
this repetition for instance can be avoided with a function that gets called for each side (left/right)